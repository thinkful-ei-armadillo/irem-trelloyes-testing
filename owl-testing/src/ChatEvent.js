import React from 'react';

function ChatEvent(props){
    return (
        <div className="ChatEvent">
            <p>{props.type}</p>
            <p>{props.message}</p>
            <p>{props.timestamp}</p>
        </div>
    );
}

export default ChatEvent;