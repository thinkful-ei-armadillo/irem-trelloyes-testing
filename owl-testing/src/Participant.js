import React from 'react';

function Participant(props){
    return (
        <div className="Participant">
            <img src={props.avatar} alt="headshot"/>
            <h3>{props.name}</h3>
            <p>{props.onStage}</p>
            <p>{props.id}</p>
        </div>
    );
}

export default Participant;