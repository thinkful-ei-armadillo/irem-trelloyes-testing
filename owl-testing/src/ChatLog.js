import React, {Component} from 'react';
import Participant from './Participant';
import store from './store';
import chatEvents from './ChatEvents';
import ChatEvent from './ChatEvent';

class ChatLog extends Component{
  static defaultProps =  { 
      participants: store,
      chatEvents: chatEvents
};

  render() {
      console.log(this.props);
    return (
        <div className="Chat-list">
          {this.props.chatEvents.map(chatEvent=> (
            <div className="ChatEvent-lis">
              <Participant 
                name={this.props.participants.find(participant => participant.id === chatEvent.participantId).name}
                avatar={this.props.participants.find(participant => participant.id === chatEvent.participantId).avatar} />
              <ChatEvent 
              type={chatEvent.type} 
              message={chatEvent.message} 
              timestamp= {new Date(chatEvent.timestamp).toString()} />
          </div>
          ))}
      </div>
  );
}
}


export default ChatLog;