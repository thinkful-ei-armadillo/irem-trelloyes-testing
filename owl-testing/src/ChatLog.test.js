import React from 'react';
import ReactDOM from 'react-dom';
import Participant from './Participant';
import ChatEvent from './ChatEvent';
import ChatLog from './ChatLog';
import renderer from 'react-test-renderer';


describe('<App />', () => {
    const participants=[
      {avatar: "https://ia.media-imdb.com/images/M/MV5BMTY5NzE3NzU3MF5BMl5BanBnXkFtZTgwMjg0NTQ5MDE@._V1_UX214_CR0,0,214,317_AL_.jpg",
      name: "Sally Salonga",
      type: "message",
      message: "Testing Chat Log",
      timestamp: "1548852484247"},
      {avatar: "https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1",
      name: "Joe Turner",
      type: "thumbs up",
      timestamp: "1548852544247"},
  ];
it('renders without crashing', () => {
  const div = document.createElement('div');

  ReactDOM.render(<ChatLog />, div);

  ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected', () => {
    const tree = renderer
      .create(<ChatLog />)
      .toJSON();
    expect(tree).toMatchSnapshot();  
    })
})