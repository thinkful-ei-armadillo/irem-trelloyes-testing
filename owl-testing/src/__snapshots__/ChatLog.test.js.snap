// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders the UI as expected 1`] = `
<div
  className="Chat-list"
>
  <div
    className="ChatEvent-lis"
  >
    <div
      className="Participant"
    >
      <img
        alt="headshot"
        src="https://robohash.org/utlaborumfugit.jpg?size=200x200&set=set1"
      />
      <h3>
        Axe Kubicka
      </h3>
      <p />
      <p />
    </div>
    <div
      className="ChatEvent"
    >
      <p>
        message
      </p>
      <p>
        Hello world
      </p>
      <p>
        Wed Jan 30 2019 07:48:04 GMT-0500 (Eastern Standard Time)
      </p>
    </div>
  </div>
  <div
    className="ChatEvent-lis"
  >
    <div
      className="Participant"
    >
      <img
        alt="headshot"
        src="https://robohash.org/doloredolorescupiditate.jpg?size=200x200&set=set1"
      />
      <h3>
        Caty Flucker
      </h3>
      <p />
      <p />
    </div>
    <div
      className="ChatEvent"
    >
      <p>
        thumbs-up
      </p>
      <p />
      <p>
        Wed Jan 30 2019 07:48:04 GMT-0500 (Eastern Standard Time)
      </p>
    </div>
  </div>
  <div
    className="ChatEvent-lis"
  >
    <div
      className="Participant"
    >
      <img
        alt="headshot"
        src="https://robohash.org/etexercitationemassumenda.jpg?size=200x200&set=set1"
      />
      <h3>
        Frank Runciman
      </h3>
      <p />
      <p />
    </div>
    <div
      className="ChatEvent"
    >
      <p>
        thumbs-down
      </p>
      <p />
      <p>
        Wed Jan 30 2019 07:48:04 GMT-0500 (Eastern Standard Time)
      </p>
    </div>
  </div>
  <div
    className="ChatEvent-lis"
  >
    <div
      className="Participant"
    >
      <img
        alt="headshot"
        src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
      />
      <h3>
        Koren Templeton
      </h3>
      <p />
      <p />
    </div>
    <div
      className="ChatEvent"
    >
      <p>
        raise-hand
      </p>
      <p />
      <p>
        Wed Jan 30 2019 07:49:04 GMT-0500 (Eastern Standard Time)
      </p>
    </div>
  </div>
  <div
    className="ChatEvent-lis"
  >
    <div
      className="Participant"
    >
      <img
        alt="headshot"
        src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
      />
      <h3>
        Koren Templeton
      </h3>
      <p />
      <p />
    </div>
    <div
      className="ChatEvent"
    >
      <p>
        clap
      </p>
      <p />
      <p>
        Wed Jan 30 2019 07:49:04 GMT-0500 (Eastern Standard Time)
      </p>
    </div>
  </div>
  <div
    className="ChatEvent-lis"
  >
    <div
      className="Participant"
    >
      <img
        alt="headshot"
        src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
      />
      <h3>
        Koren Templeton
      </h3>
      <p />
      <p />
    </div>
    <div
      className="ChatEvent"
    >
      <p>
        join
      </p>
      <p />
      <p>
        Wed Jan 30 2019 07:49:04 GMT-0500 (Eastern Standard Time)
      </p>
    </div>
  </div>
  <div
    className="ChatEvent-lis"
  >
    <div
      className="Participant"
    >
      <img
        alt="headshot"
        src="https://robohash.org/iustodoloremqueinventore.jpg?size=200x200&set=set1"
      />
      <h3>
        Ashla Attwool
      </h3>
      <p />
      <p />
    </div>
    <div
      className="ChatEvent"
    >
      <p>
        leave
      </p>
      <p />
      <p>
        Wed Jan 30 2019 07:50:04 GMT-0500 (Eastern Standard Time)
      </p>
    </div>
  </div>
  <div
    className="ChatEvent-lis"
  >
    <div
      className="Participant"
    >
      <img
        alt="headshot"
        src="https://robohash.org/utlaborumfugit.jpg?size=200x200&set=set1"
      />
      <h3>
        Axe Kubicka
      </h3>
      <p />
      <p />
    </div>
    <div
      className="ChatEvent"
    >
      <p>
        join-stage
      </p>
      <p />
      <p>
        Wed Jan 30 2019 07:51:04 GMT-0500 (Eastern Standard Time)
      </p>
    </div>
  </div>
  <div
    className="ChatEvent-lis"
  >
    <div
      className="Participant"
    >
      <img
        alt="headshot"
        src="https://robohash.org/utlaborumfugit.jpg?size=200x200&set=set1"
      />
      <h3>
        Axe Kubicka
      </h3>
      <p />
      <p />
    </div>
    <div
      className="ChatEvent"
    >
      <p>
        leave-stage
      </p>
      <p />
      <p>
        Wed Jan 30 2019 07:52:04 GMT-0500 (Eastern Standard Time)
      </p>
    </div>
  </div>
</div>
`;
